Packs/Zscaler/Integrations/Zscaler/Zscaler.py

```diff
@@ -1016,13 +1017,220 @@ def set_user_command(args):
         return responseJson
 
 
+def create_ip_destination_group(args: dict):
+    payload = dict()
+    payload['name'] = args.get('name', '')
+    payload['type'] = args.get('type', '')
+    payload['countries'] = argToList(args.get('countries', ''))
+    payload['ipCategories'] = argToList(args.get('ip_categories', ''))
+    payload['description'] = args.get('description', '')
+    payload['addresses'] = argToList(args.get('addresses', ''))
+    payload['isNonEditable'] = args.get('is_non_editable', False)
+    try:
+        cmd_url = "/ipDestinationGroups"
+        response = http_request('POST', cmd_url, data=json.dumps(payload),
+                                headers=DEFAULT_HEADERS).json()
+    except json.decoder.JSONDecodeError as exp:
+        return_error(f'Failed to execute zscaler-create-ip-destination-group command. Error: {str(exp)}')
+    content = {
+        'ID': int(response.get('id', '')),
+        'Name': response.get('name', ''),
+        'Type': response.get('type', ''),
+        'Description': response.get('description', ''),
+        'Addresses': response.get('addresses', []),
+        'IpCategories': response.get('ipCategories', []),
+        'Countries': response.get('countries', []),
+        'IsNonEditable': response.get('isNonEditable', False)
+    }
+    entry = {
+        'Type': entryTypes['note'],
+        'Contents': content,
+        'ContentsFormat': formats['json'],
+        'ReadableContentsFormat': formats['markdown'],
+        'HumanReadable': tableToMarkdown("IP Destination group created",
+                                         content),
+        'EntryContext': {'Zscaler.IPDestinationGroup': content},
+    }
+    return entry
+
+
+def list_ip_destination_groups(args: dict):
+    ip_group_id = str(args.get('ip_group_id', '')).strip()
+    exclude_type = str(args.get('exclude_type', '')).strip()
+    include_ipv6 = argToBoolean(args.get('include_ipv6', False))
+    if ip_group_id == '':
+        if include_ipv6:
+            try:
+                ipv4_cmd_url = "/ipDestinationGroups"
+                ipv6_cmd_url = "/ipDestinationGroups/ipv6DestinationGroups"
+                if exclude_type != '':
+                    ipv4_cmd_url += f"?excludeType={exclude_type}"
+                    ipv6_cmd_url += f"?excludeType={exclude_type}"
+                ipv4_responses = http_request('GET', ipv4_cmd_url).json()
+                ipv6_responses = http_request('GET', ipv6_cmd_url).json()
```

Comment:
Only these statements can result in `JSONDecodeError`, therefore these should only be inside the `try` block. Previous statements are not required to be inside `try` block.

💡 Good practice is to keep the try block as small as possible: It's best to keep only the code that might raise an exception inside the try block. This makes it easier to locate the source of the error when an exception is raised.

Reply:
done